experimental/oak_baremetal_loader/src/lookup.rs

```diff
@@ -0,0 +1,53 @@
+//
+// Copyright 2022 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+use hashbrown::HashMap;
+use oak_baremetal_communication_channel::schema;
+use std::fs;
+
+pub fn encode_lookup_data<'a>(
+    mut data: HashMap<Vec<u8>, Vec<u8>>,
+) -> oak_idl::utils::Message<schema::LookupData<'a>> {
+    let mut builder = oak_idl::utils::MessageBuilder::default();
+    let entries: Vec<flatbuffers::WIPOffset<schema::LookupDataEntry>> = data
+        .drain()
+        .map(|(key, value)| {
+            let key = builder.create_vector::<u8>(&key);
+            let value = builder.create_vector::<u8>(&value);
+            schema::LookupDataEntry::create(
+                &mut builder,
+                &schema::LookupDataEntryArgs {
+                    key: Some(key),
+                    value: Some(value),
+                },
+            )
+        })
+        .collect();
+    let items = builder.create_vector(&entries);
+    let message =
+        schema::LookupData::create(&mut builder, &schema::LookupDataArgs { items: Some(items) });
+    builder
+        .finish(message)
+        .expect("errored when creating lookup data update message")
```

Comment:
Could the function return Result instead? Perhaps returning an `oak_idl::Status` in case of error?

Reply:
Thanks, updated for it to return a result. The status code is a code idea but probably not quite right here, since this code is related to initiating a call to an IDL method, instead of handling one.